dotCMS/src/main/java/com/dotcms/analytics/track/RequestMatcher.java

```diff
@@ -0,0 +1,126 @@
+     * @return boolean true if the request match the patterns and methods
+     */
+    default boolean match(final HttpServletRequest request, final HttpServletResponse response) {
+
+        final Set<String> patterns = getMatcherPatterns();
+        final Set<String> methods  = getAllowedMethods();
+        return  Objects.nonNull(patterns) && !patterns.isEmpty() &&
+                Objects.nonNull(methods) &&  !methods.isEmpty()  &&
+                isAllowedMethod (methods, request.getMethod())   &&
+                match(request, response, patterns);
+    }
+
+    /**
+     * Match the request with the patterns
+     * @param request {@link HttpServletRequest}
+     * @param response {@link HttpServletResponse}
+     * @param patterns Set of patterns
+     * @return boolean true if any of the patterns match the request
+     */
+    default boolean match(final HttpServletRequest request, final HttpServletResponse response, final Set<String> patterns) {
+
+        final String requestURI = request.getRequestURI();
+        return patterns.stream().anyMatch(pattern -> match(requestURI, pattern));
+    }
+
+    /**
+     * Means the matcher should run before request
+     * @return boolean
+     */
+    default boolean runBeforeRequest() {
+        return false;
+    }
+
+    /**
+     * Means the matcher should run after request
+     * @return boolean
+     */
+    default boolean runAfterRequest() {
+        return false;
+    }
+    /**
+     * Match the request URI with the pattern
+     * @param requestURI String
+     * @param pattern String
+     * @return boolean true if the pattern match the request URI
+     */
+    default boolean match (final String requestURI, final String pattern) {
+
+        String uftUri = null;
+
+        try {
+
+            uftUri = URLDecoder.decode(requestURI, CHARSET);
+        } catch (UnsupportedEncodingException e) {
+
+            uftUri = requestURI;
+        }
+
+        return RegEX.containsCaseInsensitive(uftUri, pattern.trim());
+    } // match.
+
+    /**
+     * Determinate if the method is allowed
+     * @param methods Set of methods
+     * @param method String current request method
+     * @return boolean true if the method is allowed
+     */
+    default boolean isAllowedMethod(final Set<String> methods, final String method) {
+
+            return methods.contains(method);
+    }
+
+    /**
+     * Returns a set of patterns for the matcher
+     * @return Set by default empty
+     */
+    default Set<String> getMatcherPatterns() {
+
+        return Set.of();
+    }
+
+    /**
+     * Returns the request methods allowed for this matcher.
+     * @return Set by default empty
+     */
+    default Set<String> getAllowedMethods() {
+
+        return Set.of();
+    }
+
+    /**
+     * Return an id for the Matcher, by default returns the class name.
+     * @return
+     */
+    default String getId() {
+
+        return this.getClass().getName();
+    }
+}
```

Comment:
I suppose that all this default implementation is going to be used in cases like API and GraphQL, I think is better to keep this interface clean and move this default implementation to a sub interface or a class that implement then and later the Implementation for API and GraphQL can extends or implemented it.

Although it is just an idea

Reply:
Not sure if the API will use the matcher, matcher are mostly to catch requests, not api call; we suppose to create an annotation for that regard!